docs/taskruns.md

```diff
@@ -422,6 +422,86 @@ For more information, see the following topics:
 - For a list of supported `Volume` types, see [Specifying `VolumeSources` in `Workspaces`](workspaces.md#specifying-volumesources-in-workspaces).
 - For an end-to-end example, see [`Workspaces` in a `TaskRun`](../examples/v1beta1/taskruns/workspace.yaml).
 
+#### Propagated Workspaces
+
+**([alpha only](https://github.com/tektoncd/pipeline/blob/main/docs/install.md#alpha-features))**
+
+When using an embedded spec, workspaces from the parent `TaskRun` will be
+propagated to any inlined specs without needing to be explicitly defined. This
+allows authors to simplify specs by automatically propagating top-level
+workspaces down to other inlined resources.
+**Workspace substutions will only be made for `commands`, `args` and `script` fields of `steps`, `stepTemplates`, and `sidecars`.**
+
+```yaml
+apiVersion: tekton.dev/v1beta1 
+kind: TaskRun                  
+metadata:
+  generateName: propagating-workspaces-
+spec:
+  taskSpec:                    
+    steps:                     
+      - name: simple-step      
+        image: ubuntu          
+        command:               
+          - echo               
+        args:                  
+          - $(workspaces.tr-workspace.path)
+  workspaces:                  
+  - emptyDir: {}               
+    name: tr-workspace
+```
+
+Upon execution, the workspaces will be interpolated during resolution.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: propagating-workspaces-ndxnc
+  ...
+spec:
+  serviceAccountName: default
+  taskSpec:
+    steps:
+    - args:
+      - $(workspaces.tr-workspace.path)
+      command:
+      - echo
+      image: ubuntu
+      name: simple-step
+      resources: {}
+  timeout: 1h0m0s
+  workspaces:
+  - emptyDir: {}
+    name: tr-workspace
+status:
+  completionTime: "2022-09-09T17:34:22Z"
+  conditions:
+  - lastTransitionTime: "2022-09-09T17:34:22Z"
+    message: All Steps have completed executing
+    reason: Succeeded
+    status: "True"
+    type: Succeeded
+  podName: propagating-workspaces-ndxnc-pod
+  startTime: "2022-09-09T17:34:17Z"
+  steps:
+  ...
+  taskSpec:
+    steps:
+    - args:
+      - /workspace/tr-workspace
+      command:
+      - echo
+      image: ubuntu
+      name: simple-step
+      resources: {}
+    workspaces:
+    - name: tr-workspace
+
+```
+
+Workspaces can be propagated to embedded task specs, not referenced Tasks. For more information, see [Referenced TaskRuns within Embedded PipelineRuns](pipelineruns.md#referenced-taskruns-within-embedded-pipelineruns).
```

Comment:
I'd avoid linking to the PipelineRun docs from the TaskRun docs, since TaskRun docs should be able to stand alone. Can you instead provide an example here of an invalid propagation to a referenced Task?

Reply:
Yes, Can do!

Reply:
done